core/src/main/java/org/elasticsearch/common/logging/ESLogger.java

```diff
@@ -100,22 +100,14 @@
     void info(String msg, Throwable cause, Object... params);
 
     /**
-     * Logs a WARN level message.
-     */
-    void warn(String msg, Object... params);
-
-    /**
-     * Logs a WARN level message.
+     * Logs a WARN level message.  If there is any exception associated with this meessage, pass it as as the <code>cause</code>.
+     * Ohterwise, pass <code>null</code>.
```

Comment:
Maybe use `@param` for cause?  Maybe add some explanation like:
```
* @param cause exception that caused this warning. If no exception caused it pass null. Its intentional that there isn't an overload of this method without this parameter. We want to make not logging the exception inconvenient because logging exception makes issues so much easier to debug.
```